jaeger-core/src/main/java/com/uber/jaeger/samplers/RemoteControlledSampler.java

```diff
@@ -86,36 +88,31 @@ private void updateSampler() {
       return;
     }
 
-    Sampler newSampler;
-    try {
-      newSampler = extractSampler(response);
-    } catch (UnknownSamplingStrategyException e) {
-      metrics.samplerParsingFailure.inc(1);
-      return; // sampler updateGauge without a new sampler
-    }
+    Sampler oldSampler = sampler;
```

Comment:
What I mean is that you're reading `sampler` member variable multiple times and making decisions between those reads, so you have a race condition.

It may not be critical because update() should only be called internally by a single thread. Worth a comment somewhere.